src/main/java/com/ldtteam/structurize/Structurize.java

```diff
@@ -42,9 +46,23 @@ public Structurize()
         Mod.EventBusSubscriber.Bus.FORGE.bus().get().register(EventSubscriber.class);
         DistExecutor.unsafeRunWhenOn(Dist.CLIENT, () -> () -> Mod.EventBusSubscriber.Bus.FORGE.bus().get().register(ClientEventSubscriber.class));
 
-        if (DataVersion.CURRENT == DataVersion.UPCOMING)
+        if (DataFixerUtils.isVanillaDF)
         {
-            throw new RuntimeException("Missing some newest data versions. Please update com/ldtteam/structures/blueprints/v1/DataVersion");
+            if (DataFixesManager.getDataFixer().getSchema(Integer.MAX_VALUE - 10).getVersionKey() >= DataVersion.UPCOMING.getDataVersion())
+            {
+                throw new RuntimeException("You are trying to run old mod on much newer vanilla. Missing some newest data versions. Please update com/ldtteam/structures/blueprints/v1/DataVersion");
```

Comment:
```suggestion
                throw new RuntimeException("You are trying to run on a much newer game version than supported by this version of the mod. Missing some newest data versions, please update com/ldtteam/structures/blueprints/v1/DataVersion");
```
Could also remove the part that mentions the data versions and class, and just a generic "please update this mod." message (since it may confuse users).